### PR TITLE
Add failure counter for server spans

### PR DESCRIPTION
**Pull request by manishapme**
There has been an ongoing conversation in https://reddit.atlassian.net/browse/IO-2820 asking for error counts. This exists for the clients, this PR should add them for the server spans as well.

### CODE REVIEW

**Comment by spladug** [comment]
(you can test this out locally with that NullTransport / --debug stuff we talked about in OH)

**Comment by manishapme** [comment]
How do I get my local Thing vagrant to use my new version of baseplate? do I add something like `config.vm.synced_folder  ".", "/home/vagrant/baseplate"` to my `reddit-service-thing/Vagrantfile`?

**Comment by spladug** [comment]
Yeah, that sync should work and then you can `python setup.py develop` in baseplate to get it in place.

**Comment by manishapme** [comment]
💇

File: baseplate/diagnostics/metrics.py [hunk 1]
```diff
@@ -39,13 +39,16 @@ def on_server_span_created(self, context, server_span):
 class MetricsServerSpanObserver(SpanObserver):
     def __init__(self, batch, server_span):
         self.batch = batch
+        self.base_name = "server." + server_span.name
         self.timer = batch.timer("server." + server_span.name)
```
Comment (spladug) [inline baseplate/diagnostics/metrics.py]: now that `self.base_name` exists, it'd be good to use it here in the timer creation

**Comment by manishapme** [comment]
💇

File: baseplate/metrics.py [hunk 2]
```diff
@@ -95,7 +95,11 @@ def flush(self):
         # TODO: run-length compression
```
Comment (spladug) [inline baseplate/metrics.py]: i think these TODOs can be deleted now?

File: baseplate/metrics.py [hunk 3]
```diff
@@ -95,7 +95,11 @@ def flush(self):
...
         metrics, self.buffer = self.buffer, []
         message = b"\n".join(metrics)
-        self.transport.send(message)
+        try:
+            self.transport.send(message)
+        except socket.error as e:
+            logger.error("BufferedTransport flush exception, length %d, %s",
```
Comment (spladug) [inline baseplate/metrics.py]: This log message would probably be pretty confusing for someone who isn't in this code seeing it in their service's logs. Something more actionable for them would be helpful. Maybe like `baseplate metrics batch too large: flush more often or reduce amount done in one request"?

**Comment by manishapme** [comment]
💇

**Comment by manishapme** [comment]
💇  very clear message. done! 

i'll work with Courtney to deploy following instructions here: https://reddit.atlassian.net/wiki/spaces/IO/pages/98992244/Releasing+and+deploying+a+new+version+of+Baseplate when ok.

**Pull request merged**